libs/lib_parsing/Tok.ml

```diff
@@ -122,7 +122,7 @@ type t =
 and virtual_location = location * int
 [@@deriving show { with_path = false }, eq, ord, sexp]
 
-type t_always_equal = t [@@deriving show]
+type t_always_equal = t [@@deriving show, sexp, ord]
```

Comment:
why sexp and ord?
I'd rather keep things to the minimum unless absolutely needed.
For this type we actually don't want to compare them ... so definitely not 'ord'.

Reply:
I don't want ord here, not sexp. I'll remove it in another PR.

Reply:
There's a reason for `sexp`, actually. We want `t_always_equal` for `Parsed_int` because otherwise it will make equality of the Generic AST incorrect (which we want to be location-agnostic).

Reply:
(and `semgrep-pro` requires that the `Type.t` type, which relies on `Parsed_int`, be able to derive `sexp`)

Reply:
grrr, @nmote probably added deriving sexp? Would be good then to comment in AST_generic.ml and Tok.ml why we derive sexp.
There is a big comment at the top of AST_generic.ml about "Some notes on deriving", please can you update it to add sexp in the list and explain why we need it (and why we could not do it without).

Reply:
And also it looks like sexp is used only in AST_generic.ml in a few places, like for sid and alternate_name, so would be great to explain why we need it for those, but not for all the others (expr, stmt, etc.)

Reply:
I would be strongly in favor of removing it from type sid in AST_generic.ml, I don't know if it was a mistake or 
intended, but if intended it needs a big explanation